main.go

```diff
@@ -128,10 +341,161 @@ func determineShouldFetch(lastFetchedFilepath string) bool {
+			}
+			allTagSemVers = append(allTagSemVers, tagSemVer) 
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var latestReleaseTagSemVer *semver.Version
+	if len(allTagSemVers) == 0 {
+		latestReleaseTagSemVer, err = semver.StrictNewVersion(noPreviousVersion)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred creating '%s' semantic version.", noPreviousVersion)
+		}
+	} else {
+		sort.Sort(sort.Reverse(semver.Collection(allTagSemVers)))
+		latestReleaseTagSemVer = allTagSemVers[0]
+	}
+
+	return latestReleaseTagSemVer, nil
+}
+
+func doBreakingChangesExist(changelogFilepath string) (bool, error) {
+	changelogFile, err := os.Open(changelogFilepath)
+	if err != nil {
+		return false, stacktrace.Propagate(err, "An error occurred attempting to open changelog file at provided path. Are you sure '%s' exists?", changelogFilepath)
+	}
+	defer changelogFile.Close()
+
+    scanner := bufio.NewScanner(changelogFile)
+	// Find TBD header
+    for scanner.Scan() {
+		if tbdHeaderRegex.Match(scanner.Bytes()) {
+			break
+		}
+    }
+	if err := scanner.Err(); err != nil {
+        return false, stacktrace.Propagate(err, "An error occurred while scanning for the TBD header in the changelog file at provided path: '%s'.\n", changelogFilepath)
+    }
+	// Scan file until next version header detected, searching for Breaking Changes header along the way
+	foundBreakingChanges := false
+	for scanner.Scan() {
+		if breakingChangesRegex.Match(scanner.Bytes()){
+			foundBreakingChanges = true
+		}
+		if versionHeaderRegex.Match(scanner.Bytes()){
+			break
+		}
+	}
+    if err := scanner.Err(); err != nil {
+		return false, stacktrace.Propagate(err, "An error occurred while scanning for the breaking changes header in the changelog file at provided path: '%s'.\n", changelogFilepath)
+    }
+
+    return foundBreakingChanges, nil
+}
+
+func runPreReleaseScripts(preReleaseScriptsDirpath string, releaseVersion string) error {
+	preReleaseScriptsFilepath := path.Join(preReleaseScriptsDirpath, preReleaseScriptsFilename)
+	preReleaseScriptsFile, err := os.ReadFile(preReleaseScriptsFilepath);
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred attempting to open file at provided path. Are you sure '%s' exists?", preReleaseScriptsFilepath)
+	}
+
+	lines := bytes.Split(preReleaseScriptsFile, []byte("\n"))
+	var allScriptFilepaths []string
+	for _, line := range(lines) {
+		allScriptFilepaths = append(allScriptFilepaths, string(line))
+	}
+	for _, scriptFilepath := range allScriptFilepaths {
+		if strings.TrimSpace(scriptFilepath) == "" {
+			continue
+		}
+		scriptCmdString := path.Join(preReleaseScriptsDirpath, scriptFilepath)
+		scriptCmd := exec.Command(scriptCmdString, releaseVersion)
+		err := scriptCmd.Run()
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred attempting to run the following pre release script command: '%s %s'", scriptCmdString, releaseVersion)
```

Comment:
1) Same `''`/`:` mixing
2) I'd strongly recommend printing out error code and log output of the command, else the user won't know what failed, a la:
```
Pre-release script command '%s %s' returned error code '%d' and logs:\n%d
```
(this is where you'd use `:\n`, because we expect the logs to have newlines)

Reply:
You can see a solid way to do this like so: https://github.com/kurtosis-tech/kurtosis-core/blob/develop/files_artifacts_expander/main.go#L137